src/sentry/celery.py

```diff
@@ -44,6 +46,12 @@ def holds_bad_pickle_object(value, memo=None):
     if app_module.startswith(("sentry.", "getsentry.")):
         return value, "do not pickle custom classes"
 
+    if os.getenv("TASK_TYPE_CHECKING") and os.getenv("TASK_TYPE_CHECKING") != "0":
+        if isinstance(value, SafeString):
+            # Django string wrappers json encode fine
+            return None
+        elif app_module != "builtins":
+            return value, "do not pickle custom classes"
```

Comment:
Once we get all of the current failures fixed, we can mainline this to prevent future pickle parameters from creeping in.